File: app/src/main/java/com/lannix/netfilmdata/stuff/recycleView/MyRecyclerViewAdapter.java
package com.lannix.netfilmdata.stuff.recycleView;

import android.content.Context;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.CheckBox;
import android.widget.TextView;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;

import com.lannix.netfilmdata.R;

import java.util.ArrayList;

public class MyRecyclerViewAdapter extends RecyclerView.Adapter<MyRecyclerViewAdapter.MyRecycleViewHolder> {

    private LayoutInflater inflater;
    private ArrayList<ListObject> data;

    public MyRecyclerViewAdapter(Context context, ArrayList<ListObject> data) {
        inflater = LayoutInflater.from(context);
        this.data = data;
        setHasStableIds(true);
    }

    @NonNull
    @Override
    public MyRecycleViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        View view = inflater.inflate(R.layout.recycle_list_object, parent, false);
        return new MyRecycleViewHolder(view);
    }

    @Override
    public void onBindViewHolder(@NonNull MyRecycleViewHolder holder, int position) {
        holder.fill(data.get(position));
    }

    @Override
    public int getItemCount() {
        return data.size();
    }

    public static class MyRecycleViewHolder extends RecyclerView.ViewHolder {
        private TextView topicTextView;
        private TextView informationTextView;
        private CheckBox checkBox;

        public MyRecycleViewHolder(@NonNull View itemView) {
            super(itemView);
            topicTextView = itemView.findViewById(R.id.TopicTextView);
            informationTextView = itemView.findViewById(R.id.InformationTextView);
            checkBox = itemView.findViewById(R.id.CheckBox);
        }

        public void fill(ListObject listObject) {
            topicTextView.setText(listObject.topicText);
            informationTextView.setText(listObject.informationText);
        }

        public TextView getTopicTextView() {
            return topicTextView;
        }

        public TextView getInformationTextView() {
            return informationTextView;
        }

        public CheckBox getCheckBox() {
            return checkBox;
        }
    }
}

File: app/src/main/java/com/lannix/netfilmdata/stuff/recycleView/MyRecyclerViewAdapter.kt
package com.lannix.netfilmdata.stuff.recycleView

import android.content.Context
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.CheckBox
import android.widget.TextView
import androidx.recyclerview.widget.RecyclerView

import com.lannix.netfilmdata.R

import java.util.ArrayList

class MyRecyclerViewAdapter(context: Context, private val data: ArrayList<ListObject>) : RecyclerView.Adapter<MyRecyclerViewAdapter.MyRecycleViewHolder>() {

    private val inflater: LayoutInflater

    init {
        inflater = LayoutInflater.from(context)
        setHasStableIds(true)
    }

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): MyRecycleViewHolder {
        val view = inflater.inflate(R.layout.recycle_list_object, parent, false)
        return MyRecycleViewHolder(view)
    }

    override fun onBindViewHolder(holder: MyRecycleViewHolder, position: Int) {
        holder.fill(data[position])
    }

    override fun getItemCount(): Int {
        return data.size
    }

    class MyRecycleViewHolder(itemView: View) : RecyclerView.ViewHolder(itemView) {
        val topicTextView: TextView
        val informationTextView: TextView
        val checkBox: CheckBox

        init {
            topicTextView = itemView.findViewById(R.id.TopicTextView)
            informationTextView = itemView.findViewById(R.id.InformationTextView)
            checkBox = itemView.findViewById(R.id.CheckBox)
        }

        fun fill(listObject: ListObject) {
            topicTextView.text = listObject.topicText
            informationTextView.text = listObject.informationText
        }
    }
}
